libs/api/bpf_syscall.cpp

```diff
@@ -6,188 +6,230 @@
 #include <windows.h>
 #include <WinError.h>
 #include <linux/bpf.h>
+#include <sal.h>
+#include <stdexcept>
 
-#define CHECK_SIZE(last_field_name)                                                    \
-    if (!tail_is_zero(                                                                 \
-            attr,                                                                      \
-            offsetof(union bpf_attr, last_field_name) + sizeof(attr->last_field_name), \
-            sizeof(union bpf_attr))) {                                                 \
-        return -EINVAL;                                                                \
-    }
-
-static bool
-tail_is_zero(const void* buf, unsigned int start, unsigned int end)
+template <typename T> class ExtensibleStruct
 {
-    const unsigned char* p = (const unsigned char*)buf + start;
-    const unsigned char* e = (const unsigned char*)buf + end;
+  private:
+    void* _source;
+    size_t _copy_size;
+    T _tmp;
+    T* _ptr;
```

Comment:
```suggestion
    T _temporary;
    T* _pointer;
```
Unless the name needs to match something outside this template